### ISSUE TITLE
Can't create agent when no plugins are specified

### ISSUE THREAD

**Issue by carolynvs** [getporter/operator#165]
When no plugins are specified in the AgentConfigs, the following error occurs in the AgentAction controller:

```
2023-02-24T18:59:46Z	ERROR	Reconciler error	{"controller": "agentaction", "controllerGroup": "getporter.org", "controllerKind": "AgentAction", "AgentAction": {"name":"default","namespace":"ginkgo-tests-94672"}, "namespace": "ginkgo-tests-94672", "name": "default", "reconcileID": "9362a146-b03c-4f99-b42e-83013d9b34aa", "error": "error creating Porter agent job: Job.batch \"default-ktb6j\" is invalid: [spec.template.spec.volumes[3].persistentVolumeClaim.claimName: Required value, spec.template.spec.containers[0].volumeMounts[3].name: Not found: \"porter-plugins\"]", "errorVerbose": "Job.batch \"default-ktb6j\" is invalid: [spec.template.spec.volumes[3].persistentVolumeClaim.claimName: Required value, spec.template.spec.containers[0].volumeMounts[3].name: Not found: \"porter-plugins\"]\nerror creating Porter agent job\nget.porter.sh/operator/controllers.(*AgentActionReconciler).createAgentJob\n\t/workspace/controllers/agentaction_controller.go:468\nget.porter.sh/operator/controllers.(*AgentActionReconciler).runPorter\n\t/workspace/controllers/agentaction_controller.go:214\nget.porter.sh/operator/controllers.(*AgentActionReconciler).Reconcile\n\t/workspace/controllers/agentaction_controller.go:95\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:122\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:323\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:274\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:235\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1594"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:329
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:274
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:235
```

The problem is that it unconditionally looks up the name of the porter-plugins volume to mount into the agent and applies it to the job spec. When no plugins are specified, we don't need to mount a volume and the volume name is `""`.

I tried modifying the operator code to skip mounting when the volume name is empty, and that fixed the problem but I'm wondering if it should always mount the k8s plugin when none are specified? I think this may be a bug in our defaulting and the porter agent shouldn't run without plugins.

**Comment by carolynvs** [comment]
I'm going to fix this issue with the fix mentioned in the OP but @VinozzZ when you get a chance can you think about if in this case we should have defaulted to the k8s plugin? I can create a follow-up issue for that if you think that's the correct behavior.

**Comment by VinozzZ** [comment]
Hmmm, this is not what I expected when no plugins are specified, the agent config controller should not have created a agent action? or is that still a possibility?

**Comment by carolynvs** [comment]
This is a problem in the AgentAction controller. When it creates the job for the agent action, the resolved agent config is coming back with no plugins, so the plugin pvc to mount into the agent is empty. If you see the PR, you can see where it's happening. I kind of expected that when the agent config is resolved, that the plugins would always be populated with at least the k8s plugin but instead it's empty.

**Comment by VinozzZ** [comment]
Ohhh, duh, I actually remember adding the check locally lol

I think we landed on not having a default plugin set due to some other reason. I think it's documented in one of the issues

**Comment by carolynvs** [comment]
I'll go with the fix in #166 and if we decide later that we need the default plugin, we can update it. 👍
